src/openrct2/windows/tile_inspector.c

```diff
@@ -910,9 +932,11 @@ static void window_tile_inspector_mouseup(rct_window *w, rct_widgetindex widgetI
 	case WIDX_BUTTON_CORRUPT:
 		window_tile_inspector_insert_corrupt_element(w->selected_list_item);
 		break;
-	case WIDX_BUTTON_REMOVE:
+	case WIDX_BUTTON_REMOVE: {
+		sint32 nextItemToSelect = w->selected_list_item;
```

Comment:
Since the list is reversed (first elements at the bottom), subtract 1 from here. That will prevent the selected row from moving, and look to the user as if the index is the same.